### ISSUE TITLE
storage/types: panic: runtime error: slice bounds out of range

### ISSUE THREAD

**Issue by nonsense** [ethersphere/swarm#682]
```
swarm_30402 panic: runtime error: slice bounds out of range
swarm_30402
swarm_30402 goroutine 80 [running]:
swarm_30402 github.com/ethereum/go-ethereum/swarm/storage.(*ContentAddressValidator).Validate(0xc4202747c0, 0xc427662ba0, 0x20, 0x20, 0x1a60588, 0x0, 0x0, 0xc4210553ac)
swarm_30402     /go/src/github.com/ethereum/go-ethereum/swarm/storage/types.go:344 +0x37d
swarm_30402 github.com/ethereum/go-ethereum/swarm/storage.(*LocalStore).Put(0xc421055380, 0xc4255c2280)
swarm_30402     /go/src/github.com/ethereum/go-ethereum/swarm/storage/localstore.go:102 +0xd4
swarm_30402 github.com/ethereum/go-ethereum/swarm/storage.(*DBAPI).Put(0xc421030ec0, 0xc4255c2280)
swarm_30402     /go/src/github.com/ethereum/go-ethereum/swarm/storage/dbapi.go:51 +0x39
swarm_30402 github.com/ethereum/go-ethereum/swarm/network/stream.(*Delivery).processReceivedChunks(0xc4210553b0)
swarm_30402     /go/src/github.com/ethereum/go-ethereum/swarm/network/stream/delivery.go:224 +0x19c
swarm_30402 created by github.com/ethereum/go-ethereum/swarm/network/stream.NewDelivery
swarm_30402     /go/src/github.com/ethereum/go-ethereum/swarm/network/stream/delivery.go:59 +0xbe
```

**Comment by gbalint** [comment]
It seems it can happen that a chunk received in processReceivedChunks is too short, less than 8 bytes.
I don't really know how this can happen. So as a first step I submitted a PR to log the other side of this scenario: if it happens that a short chunk is delivered, and also to know if it happens during syncing or delivering: https://github.com/ethersphere/go-ethereum/pull/688

**Comment by nonsense** [comment]
This looks relevant:

```
swarm_30401 DEBUG[06-11|14:48:43.238685|swarm/api/http/server.go:1052]       parsed request path                      ruid=299c532e method=GET uri.Addr=c33bd70b1cc6c9b06cea407704a1d9f29d63718df37d5b5366faf90098af1bcf uri.Path=                   uri.Scheme=bzz
swarm_30401 DEBUG[06-11|14:48:43.238758|swarm/api/http/server.go:907]        handle.get.file                          ruid=299c532e
swarm_30401 DEBUG[06-11|14:48:43.238807|swarm/api/http/server.go:928]        handle.get.file: resolved                ruid=299c532e key=c33bd70b1cc6c9b06cea407704a1d9f29d63718df37d5b5366faf90098af1bcf
swarm_30401 DEBUG[06-11|14:48:43.238879|swarm/api/api.go:314]                api.get                                  key=c33bd70b1cc6c9b06cea407704a1d9f29d63718df37d5b5366faf90098af1bcf path=
swarm_30401 DEBUG[06-11|14:48:43.238955|swarm/storage/chunker.go:398]        lazychunkreader.size                     key=c33bd70b1cc6c9b06cea407704a1d9f29d63718df37d5b5366faf90098af1bcf
swarm_30401 DEBUG[06-11|14:48:43.239242|swarm/storage/chunker.go:535]        lazychunkreader.read                     key=c33bd70b1cc6c9b06cea407704a1d9f29d63718df37d5b5366faf90098af1bcf
swarm_30401 DEBUG[06-11|14:48:43.239321|swarm/storage/chunker.go:398]        lazychunkreader.size                     key=c33bd70b1cc6c9b06cea407704a1d9f29d63718df37d5b5366faf90098af1bcf
swarm_30401 DEBUG[06-11|14:48:43.239392|swarm/api/manifest.go:240]           manifest retrieved                       key=c33bd70b1cc6c9b06cea407704a1d9f29d63718df37d5b5366faf90098af1bcf
swarm_30401 DEBUG[06-11|14:48:43.239494|swarm/api/api.go:324]                trie getting entry                       key=c33bd70b1cc6c9b06cea407704a1d9f29d63718df37d5b5366faf90098af1bcf path=
swarm_30401 DEBUG[06-11|14:48:43.239563|swarm/api/api.go:328]                trie got entry                           key=c33bd70b1cc6c9b06cea407704a1d9f29d63718df37d5b5366faf90098af1bcf path=                   entry.Hash=8f7de1bd36ceb8c599e18f9018b6ff04ca06033d50f6a914dda23f2b21c44f25
swarm_30401 DEBUG[06-11|14:48:43.239699|swarm/api/api.go:412]                content lookup key                       key=8f7de1bd36ceb8c599e18f9018b6ff04ca06033d50f6a914dda23f2b21c44f25 mimetype=application/octet-stream
swarm_30401 DEBUG[06-11|14:48:43.239783|swarm/storage/chunker.go:398]        lazychunkreader.size                     key=8f7de1bd36ceb8c599e18f9018b6ff04ca06033d50f6a914dda23f2b21c44f25
swarm_30401 DEBUG[06-11|14:48:43.240636|swarm/storage/chunker.go:555]        lazychunkreader.seek                     key=8f7de1bd36ceb8c599e18f9018b6ff04ca06033d50f6a914dda23f2b21c44f25 offset=0
swarm_30401 DEBUG[06-11|14:48:43.240681|swarm/storage/chunker.go:555]        lazychunkreader.seek                     key=8f7de1bd36ceb8c599e18f9018b6ff04ca06033d50f6a914dda23f2b21c44f25 offset=0
swarm_30401 DEBUG[06-11|14:48:43.240913|swarm/storage/chunker.go:535]        lazychunkreader.read                     key=8f7de1bd36ceb8c599e18f9018b6ff04ca06033d50f6a914dda23f2b21c44f25
swarm_30401 DEBUG[06-11|14:48:43.241938|swarm/storage/chunker.go:398]        lazychunkreader.size                     key=8f7de1bd36ceb8c599e18f9018b6ff04ca06033d50f6a914dda23f2b21c44f25
swarm_30401 DEBUG[06-11|14:48:53.259591|swarm/storage/netstore.go:112]       NetStore.Get retry chunk                 key=760fd80d4c75de991f775cf9d42635871b68cdd574b6ca715a85ba92a6844c94
swarm_30401 DEBUG[06-11|14:48:57.378831|p2p/dial.go:333]                     Resolving node failed                    id=f3df0add0bf384aa newdelay=2m0s
swarm_30401 DEBUG[06-11|14:48:59.124843|swarm/storage/chunker.go:535]        lazychunkreader.read                     key=8f7de1bd36ceb8c599e18f9018b6ff04ca06033d50f6a914dda23f2b21c44f25
swarm_30401 DEBUG[06-11|14:48:59.124993|swarm/storage/chunker.go:398]        lazychunkreader.size                     key=8f7de1bd36ceb8c599e18f9018b6ff04ca06033d50f6a914dda23f2b21c44f25
swarm_30401 DEBUG[06-11|14:48:59.880548|swarm/storage/chunker.go:535]        lazychunkreader.read                     key=8f7de1bd36ceb8c599e18f9018b6ff04ca06033d50f6a914dda23f2b21c44f25
swarm_30401 DEBUG[06-11|14:48:59.880669|swarm/storage/chunker.go:398]        lazychunkreader.size                     key=8f7de1bd36ceb8c599e18f9018b6ff04ca06033d50f6a914dda23f2b21c44f25
swarm_30401 DEBUG[06-11|14:49:09.893045|swarm/storage/netstore.go:112]       NetStore.Get retry chunk                 key=7f5c437823e1aee1cd8e2c5be27c7b638e147ebe0f0cb209ecde635701daefbe
swarm_30401 DEBUG[06-11|14:49:09.893358|swarm/storage/netstore.go:112]       NetStore.Get retry chunk                 key=7da918144753524ebab7b5092416cf81f01ab459c90852c24d65574019b7d483
swarm_30401 panic: runtime error: slice bounds out of range
swarm_30401
swarm_30401 goroutine 497500 [running]:
swarm_30401 github.com/ethereum/go-ethereum/swarm/storage.ChunkData.Size(...)
swarm_30401     /go/src/github.com/ethereum/go-ethereum/swarm/storage/types.go:318
swarm_30401 github.com/ethereum/go-ethereum/swarm/storage.(*LazyChunkReader).join(0xc422540e00, 0xc42671e000, 0x1000, 0x6000, 0x0, 0x1000, 0x0, 0x20, 0x0, 0x0, ...)
swarm_30401     /go/src/github.com/ethereum/go-ethereum/swarm/storage/chunker.go:472 +0x4f6
swarm_30401 github.com/ethereum/go-ethereum/swarm/storage.(*LazyChunkReader).join.func1(0xc425ca2020, 0x1008, 0x12e0, 0xc422540e00, 0xc4213b5f20, 0x40000, 0xc4205ac1f0, 0xc4213b5e60, 0xc4205ac2e8, 0xc426704000, ...)
swarm_30401     /go/src/github.com/ethereum/go-ethereum/swarm/storage/chunker.go:528 +0x20a
swarm_30401 created by github.com/ethereum/go-ethereum/swarm/storage.(*LazyChunkReader).join
swarm_30401     /go/src/github.com/ethereum/go-ethereum/swarm/storage/chunker.go:514 +0x333
```

**Comment by nonsense** [comment]
```
swarm_30399 ERROR[06-11|15:31:00.000281|swarm/storage/types.go:350]          invalid content address                  expected=807361cda383a714e2166defd56bba484f620f761a4c4aa8e734043e830e1eb1 have=545f1732d2fa61dbd0a849f176d8921c4160fcf241ad5bbb46765a70ba92749e
swarm_30399 DEBUG[06-11|15:31:00.000336|swarm/storage/types.go:351]          invalid content address data             addr=545f1732d2fa61dbd0a849f176d8921c4160fcf241ad5bbb46765a70ba92749e data=00002e000000000089951e5b000000002c010000000000006d797265736f757263657a2e393025c567ec4089d34f393ae6b5c234536a
swarm_30399 panic: runtime error: slice bounds out of range
swarm_30399
swarm_30399 goroutine 15 [running]:
swarm_30399 github.com/ethereum/go-ethereum/swarm/storage/mru.(*Handler).parseUpdate(0xc420132480, 0xc422ef2840, 0x36, 0x36, 0xc42065a510, 0xc422a5e3a0, 0xc423deb6c0, 0x6c, 0xc422ef2880, 0x40, ...)
swarm_30399     /go/src/github.com/ethereum/go-ethereum/swarm/storage/mru/resource.go:737 +0xc5c
swarm_30399 github.com/ethereum/go-ethereum/swarm/storage/mru.(*Handler).Validate(0xc420132480, 0xc4258a8780, 0x20, 0x20, 0xc422ef2840, 0x36, 0x36, 0x0)
swarm_30399     /go/src/github.com/ethereum/go-ethereum/swarm/storage/mru/resource.go:288 +0x6a
swarm_30399 github.com/ethereum/go-ethereum/swarm/storage.(*LocalStore).Put(0xc420d44b70, 0xc4235a6a80)
swarm_30399     /go/src/github.com/ethereum/go-ethereum/swarm/storage/localstore.go:102 +0xd4
swarm_30399 github.com/ethereum/go-ethereum/swarm/storage.(*DBAPI).Put(0xc420d373d0, 0xc4235a6a80)
swarm_30399     /go/src/github.com/ethereum/go-ethereum/swarm/storage/dbapi.go:51 +0x39
swarm_30399 github.com/ethereum/go-ethereum/swarm/network/stream.(*Delivery).processReceivedChunks(0xc420d44ba0)
swarm_30399     /go/src/github.com/ethereum/go-ethereum/swarm/network/stream/delivery.go:227 +0x2e6
swarm_30399 created by github.com/ethereum/go-ethereum/swarm/network/stream.NewDelivery
swarm_30399     /go/src/github.com/ethereum/go-ethereum/swarm/network/stream/delivery.go:58 +0xbe
```
